### PR TITLE
feat(gateway): Unix socket transport with TCP fallback

### LINKED ISSUE #179: Unix Socket Listener — Replace TCP for Local Daemon Mode
## Overview

Replace the TCP listener (`0.0.0.0:40257`) with a Unix domain socket for local daemon mode. Remote daemon mode continues to use TCP unchanged. The entire HTTP/WebSocket API surface is untouched — only the transport layer changes.

## Background

Part of rabbytesoftware/quiver.core#178 — see that issue for full context and architectural decisions.

### Config

Add `api.transport` to the daemon config:

```yaml
api:
  transport: socket      # "socket" for local, "tcp" for remote
  socket_path: ""        # optional override; resolved automatically if empty
  host: 0.0.0.0          # only used when transport: tcp
  port: 40257            # only used when transport: tcp
```

Default for local daemon: `socket`. Existing remote deployments set `transport: tcp` and are unaffected.

### Socket Path Resolution

Resolved automatically based on platform when `socket_path` is not set:

| Platform | Default Path |
|---|---|
| macOS | `$HOME/Library/Application Support/quiver/quiver.sock` |
| Linux | `$XDG_RUNTIME_DIR/quiver.sock` |
| Windows | `%LOCALAPPDATA%\quiver\quiver.sock` |

### Listener Swap

`internal/api/container.go` currently calls `engine.Run(addr)`. Replace with `engine.RunListener(net.Listener)`:

```go
// local mode
listener, _ := net.Listen("unix", socketPath)
os.Chmod(socketPath, 0600)

// remote mode
listener, _ := net.Listen("tcp", addr)

engine.RunListener(listener)
```

No changes anywhere else in the stack.

### Startup & Cleanup

- Create parent directory if it does not exist
- On startup: if socket file exists, attempt to connect — if refused (stale), remove and recreate
- Set permissions to `0600` immediately after creation
- On graceful shutdown: remove the socket file

### CLI Flags

`--host` and `--port` on the `daemon` command remain valid for remote mode. Add `--socket` for explicit path override in local mode.

## What Does NOT Change

- All HTTP routes, handlers, and DTOs
- WebSocket endpoints, hub, and broadcaster logic
- Middleware stack (logger, timer, recovery)
- Any remote deployment or existing client that uses TCP

## Testing

- Socket path resolution per platform
- Daemon starts, accepts connection over unix socket, all existing API endpoints respond correctly
- Stale socket file handled cleanly on startup
- Socket file removed on shutdown
- `--socket` flag override works

## Dependencies

- Part of: rabbytesoftware/quiver.core#178
- Blocks: rabbytesoftware/quiver.desktop (IPC module)

### PR DESCRIPTION
**Pull request by char2cs**
## Summary

- Adds `internal/core/gateway` package with a `Transport` interface, Unix socket implementation, and TCP fallback
- Wires the daemon to obtain a listener from the gateway rather than binding directly, making the socket path configurable via `config.yaml`
- Updates integration test kit (`tests/kit/env.go`, `client.go`, `typed_client.go`, `bench_test.go`) to route all HTTP clients through the Unix socket instead of a bare `http.Client`
- Fixes pre-existing flaky `TestList_FollowedFilter` (async follow projection race — now uses `Eventually`)
- Adds deterministic `TestDeps_CatalogCleanAfterFailedInstall` test to cover async install-failure catalog-cleanliness path
- Adds `WaitForState` guard in `TestDeps_CatalogCleanAfterFailedAdd` for the edge case where cycle detection slips to the wizard
- Patches gosec G112 (`ReadHeaderTimeout` missing on HTTP server)

## Test plan

- [ ] `make pr-checks` passes (lint + unit + integration + bench)
- [ ] `TestDeps_CatalogCleanAfterFailedInstall` goes red when `sendEndExecution` cleanup is removed, green when present
- [ ] Daemon starts and accepts connections on the Unix socket path from config

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/integration/collections/collections_test.go [hunk 1]
```diff
@@ -226,6 +226,12 @@ func (s *CollectionSuite) TestList_FollowedFilter() {
 	s.Equal(http.StatusCreated, tc.CollectionFollow(gamingNS))
 
 	trueVal := true
+	// Follow projection is async — poll until visible before asserting.
```
Comment (char2cs) [inline tests/integration/collections/collections_test.go]: Why pool? can't we wait on the websocket side of things?
